connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala

```diff
@@ -47,18 +47,29 @@ class SparkConnectStreamHandler(responseObserver: StreamObserver[ExecutePlanResp
     extends Logging {
 
   def handle(v: ExecutePlanRequest): Unit = SparkConnectArtifactManager.withArtifactClassLoader {
-    val session =
-      SparkConnectService
-        .getOrCreateIsolatedSession(v.getUserContext.getUserId, v.getSessionId)
-        .session
+    val sessionHolder = SparkConnectService
+      .getOrCreateIsolatedSession(v.getUserContext.getUserId, v.getSessionId)
+    val session = sessionHolder.session
+
     session.withActive {
+      val debugString = try {
+        Utils.redact(
+          session.sessionState.conf.stringRedactionPattern,
+          ProtoUtils.abbreviate(v).toString)
+      } catch {
+        case e: Throwable =>
```

Comment:
Catch NonFatal please? I know the old code did the same, but swallowing errors is typically bad.